app/models/manageiq/providers/ibm_cloud/vpc/cloud_manager/logging_mixin.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# Encapsulate logging methods into a single mixin.
+module ManageIQ::Providers::IbmCloud::VPC::CloudManager::LoggingMixin
+  # A logger wrapper that can be used as a general interface during dev or production.
+  # During development use the defined $ibm_cloud_log which prints to ibm_cloud.log.
+  # In production use the defined _log which prints to general evm log and should be captured in containers.
+  class LoggingWrapper
+    # Define a short-lived object for this method.
+    # @param method_name [String] The name of the method calling this logger.
+    # @return [void]
+    def initialize(method_name)
+      @method_name = method_name
+    end
+
+    # Send a info message to the configured logger.
+    # @param msg [String] A string message to print.
+    # @return [void]
+    def info(msg)
+      parent_logger.info(format_message(msg))
+    end
+
+    # Send a debug message to the configured logger.
+    # @param msg [String] A string message to print.
+    # @return [void]
+    def debug(msg)
+      parent_logger.debug(format_message(msg))
+    end
+
+    # Send a warn message to the configured logger.
+    # @param msg [String] A string message to print.
+    # @return [void]
+    def warn(msg)
+      parent_logger.warn(format_message(msg))
+    end
+
+    # Send a warn message to the configured logger.
+    # @param msg [String] A string message to print.
+    # @return [void]
+    def error(msg)
+      parent_logger.error(format_message(msg))
+    end
+
+    # Log the backtrace for an exception.
+    # @param exception [Exception] The exception object.
+    # @param context_msg [String] A message to be printed as an error that provides context for the exception.
+    # @param re_raise [Boolean] Whether to raise the original error again.
+    # @param send_return [] The object to send as a return value.
+    # @return [void]
+    def log_backtrace(exception, context_msg: '', re_raise: true, send_return: nil)
+      parent_logger.error(format_message(context_msg)) unless context_msg.length.zero?
+
+      parent_logger.log_backtrace(exception)
+      raise if re_raise
```

Comment:
This is unintuitive.  The log_backtrace method from the base VMDBLogger already has the semantics that it just logs the backtrace (as per the name) and nothing else.  Adding in this raise changes those semantics and makes it complicated for the reader (hence my eventual [comment below](https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/198/files#r624065699)).

Typically, log_backtrace is only used in places where we don't want to re-raise but need contextual information.  Otherwise letter the errors buble up is often sufficient.